### ISSUE TITLE
When using HandleInner() not all AggregateExceptions are being checked

### ISSUE THREAD

**Issue by sideproject** [App-vNext/Polly#818]
When handling exceptions that that are wrapped in an AggregateException, it looks like only the first exception is recursively searched for a matching predicate.  If there are two (or more) exceptions in the InnerExceptions collection, they are not searched.

```csharp
//Example Policy
var policy = Policy
    .HandleInner<DivideByZeroException>()
    .Retry(3, (exception, _) => passedToOnRetry = exception);
```
```csharp
//The policy will find and retry the nested DivideByZeroException
Exception willRetryException = new AggregateException(
    new Exception("First: With Inner Exception"),
        new DivideByZeroException()));
    new Exception("Second: Without Inner Exception",
```
```csharp
//The policy will NOT find and retry the nested DivideByZeroException
Exception willNotRetryException = new AggregateException(
    new Exception("First: Without Inner Exception"),
    new Exception("Second: With Inner Exception",
        new DivideByZeroException()));
```

I was able to make this work by modifying the HandleInner function in  src/Polly/PolicyBuilder.OrSyntax.cs:55.

```csharp
internal static ExceptionPredicate HandleInner(Func<Exception, bool> predicate)
{
    return exception =>
    {
        if (exception is AggregateException aggregateException)
        {
            foreach (var innerException in aggregateException.Flatten().InnerExceptions)
            {
                var matchedInAggregate = HandleInnerNested(predicate, innerException);
                if (matchedInAggregate != null)
                    return matchedInAggregate;
            }
        }

        return HandleInnerNested(predicate, exception);
    };
}
```

I'm happy to submit a pull request with tests for this if you're willing to entertain it.

**Comment by sideproject** [comment]
Just checking in - is this change something that you'd entertain?  I'm happy to submit a pull request for discussion.

**Comment by martincostello** [comment]
This _seems_ reasonable to me, but I'm not sure whether or not it could be considered a breaking behavioural change or not.

I guess if it's a minor enough code change to make you could open a PR for it and solicit further comment/opinion.

**Comment by sideproject** [comment]
Understood.  I've submitted the PR for your review.

**Comment by sideproject** [comment]
Understood, I've created PR #822 for your review.

The CI appears to have failed but doesn't seem to be related to my change - `Could not load file or assembly 'YamlDotNet,`

(Apologies for the second PR, I had some trouble with the first one so I started over)

**Comment by martincostello** [comment]
Fixed in v7.2.2.
